pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -240,12 +241,13 @@ private List<String> getInstanceIds(String query, List<String> tableNames, Strin
     List<String> instanceIds;
     if (!tableNames.isEmpty()) {
       List<TableConfig> tableConfigList = getListTableConfigs(tableNames, database);
-      if (tableConfigList == null || tableConfigList.isEmpty()) {
+      List<LogicalTableConfig> logicalTableConfigList = getListLogicalTableConfigs(tableNames, database);
+      if ((tableConfigList == null || tableConfigList.isEmpty()) && logicalTableConfigList.isEmpty()) {
```

Comment:
`tableConfigList` can only be empty right?

Reply:
in `getListTableConfig`, the following if condition is triggered if no offline or realtime table is found for one of the items in the list. Thats why a null check is required. I have added `@Nullable` annotation to `getListTableConfigs`. I dont know the reason for this behaviour.

```
      if (tableConfigList.isEmpty()) {
        return null;
      }

```

Hmm - I should do the same for logical tables ? This code needs work. The code has to throw if no table or logical table config is found for a specific entry.

Reply:
This change was more complicated than I imagined. However I have added tests in `BaseLogicalTableIntegrationTest`. Tell me if you are more comfortable if I remove the diff for controller submit. I can create a separate PR for that.

Reply:
Thanks for the additional tests. I think the logic looks fine, and I'm okay with merging it with the rest of the changes here.